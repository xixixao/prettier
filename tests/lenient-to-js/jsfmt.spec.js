function parser(text, { babylon }) {
  const x = babylon(
    text,
    {},
    {},
    {
      plugins: ["lenient", "flow", "jsx"]
    }
  );
  // console.log(JSON.stringify(x, null, 2));
  return x;
}

run_spec(__dirname, [parser], {
  bracketSpacing: false,
  semi: true
});
