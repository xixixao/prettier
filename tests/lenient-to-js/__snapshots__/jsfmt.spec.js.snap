// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrows.js 1`] = `
g = x =>
  x * square(x);

() => {}

() => ({})

() => {a: 3}

() =>
  {
    b: 3
  }

bare = () => {;}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const g = x => {
  x * square(x);
};

() => ({});

() => ({});

() => ({a: 3});

() => ({
  b: 3
});

const bare = () => {};

`;

exports[`classes.js 1`] = `
class X extends B
  other()
    da()

bla = 5
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class X extends B {
  other() {
    da();
  }
}

const bla = 5;

`;

exports[`declarations.js 1`] = `
let x := 3
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
let x = 3;

`;

exports[`destructure.js 1`] = `
const {x} = y
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const {x} = y;

`;

exports[`labels.js 1`] = `
;loop: for (;;)
  break

loop: X = 42;

b:
  | A
  | B = 1;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loop: for (;;) {
  break;
}

const loop: X = 42;

const b: A | B = 1;

`;

exports[`objects.js 1`] = `
math = {
  root: Math.sqrt
  square: square
  cube: x =>
    x * sque(x);
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const math = {
  root: Math.sqrt,
  square: square,
  cube: x => {
    x * sque(x);
  }
};

`;
